### PR TITLE
Simplify composer install in CI

### PR DESCRIPTION
**Pull request by OskarStark**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -35,21 +35,12 @@ jobs:
...
                   extensions: zip
 
-            - name: Get composer cache directory
-              id: composercache
-              run: echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
-
-            - name: Cache dependencies
-              uses: actions/cache@v4
+            - name: "Install dependencies with composer"
+              uses: "ramsey/composer-install@3.0.0"
```
Comment (dunglas) [inline .github/workflows/ci.yml:39]:
```suggestion
              uses: "ramsey/composer-install@v3"
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -74,19 +65,10 @@ jobs:
...
```
Comment (dunglas) [inline .github/workflows/ci.yml:69]:
```suggestion
              uses: "ramsey/composer-install@v3"
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -108,22 +90,13 @@ jobs:
...
-                  key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}
-                  restore-keys: ${{ runner.os }}-composer-
-
             - name: Allow dev dependencies
               run: composer config minimum-stability dev
 
-            - name: Install dependencies
-              run: composer install --prefer-dist
+            - name: "Install dependencies with composer"
+              uses: "ramsey/composer-install@3.0.0"
```
Comment (dunglas) [inline .github/workflows/ci.yml:97]:
```suggestion
              uses: "ramsey/composer-install@v3"
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -240,19 +186,10 @@ jobs:
...
             extensions: zip
 
-        - name: Get composer cache directory
-          id: composercache
-          run: echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
-
-        - name: Cache dependencies
-          uses: actions/cache@v4
+        - name: "Install dependencies with composer"
+          uses: "ramsey/composer-install@3.0.0"
```
Comment (dunglas) [inline .github/workflows/ci.yml:190]:
```suggestion
          uses: "ramsey/composer-install@v3"
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -206,19 +161,10 @@ jobs:
...
                   extensions: zip
 
-            - name: Get composer cache directory
-              id: composercache
-              run: echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
-
-            - name: Cache dependencies
-              uses: actions/cache@v4
+            - name: "Install dependencies with composer"
+              uses: "ramsey/composer-install@3.0.0"
```
Comment (dunglas) [inline .github/workflows/ci.yml:165]:
```suggestion
              uses: "ramsey/composer-install@v3"
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -176,19 +140,10 @@ jobs:
...
```
Comment (dunglas) [inline .github/workflows/ci.yml:144]:
```suggestion
              uses: "ramsey/composer-install@v3"
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -141,19 +114,10 @@ jobs:
...
```
Comment (dunglas) [inline .github/workflows/ci.yml:118]:
```suggestion
              uses: "ramsey/composer-install@v3"
```

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -35,21 +35,12 @@ jobs:
...
               with:
-                  path: ${{ steps.composercache.outputs.dir }}
-                  key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}
-                  restore-keys: ${{ runner.os }}-composer-
-
-            - name: Install dependencies
-              run: composer install --prefer-dist
+                  composer-options: "--prefer-dist"
```
Comment (dunglas) [inline .github/workflows/ci.yml:41]: This is now the default. This can be removed everywhere.

**Review by dunglas — DISMISSED** [review]

File: .github/workflows/ci.yml (hunk 8 above)
Comment (OskarStark) [reply .github/workflows/ci.yml:41]: Are you sure? I double checked the README yesterday, because I was unsure and it isn't mentioned explicitly

Comment (dunglas) [reply .github/workflows/ci.yml:41]: Yes!

> Composer uses dist by default.

https://getcomposer.org/doc/03-cli.md#install-i

Comment (OskarStark) [reply .github/workflows/ci.yml]: Ah indeed, thanks for clarification

**Review by nicolas-grekas — APPROVED** [review]

**Pull request merged**